src/main/java/org/jabref/gui/LibraryTab.java

```diff
@@ -239,11 +241,15 @@ public void feedData(BibDatabaseContext bibDatabaseContext) {
         cleanUp();
 
         this.bibDatabaseContext = Objects.requireNonNull(bibDatabaseContext);
-        // When you open an existing library, a library tab with a loading animation is added immediately.
-        // At that point, the library tab is given a temporary bibDatabaseContext with no entries.
-        // This line is necessary because, while there is already a binding that updates the active database when a new tab is added,
-        // it doesn't handle the case when a library is loaded asynchronously.
-        stateManager.setActiveDatabase(bibDatabaseContext);
+
+        if (this.getTabPane().getSelectionModel().selectedItemProperty().equals(this)) {
```

Comment:
```suggestion
        if (this.getTabPane().getSelectionModel().selectedItemProperty().get().equals(this)) {
```

Reply:
@calixtus prevented me from using `frame.getCurrentLibraryTab().equals(this)`, because he wants to get rid of `frame`. Fixed.

Reply:
Yes I did and I'm not even ashamed of it.